gitpod-io/gitpod
# Dedicated Onboarding - Adding layout/getting started view

## Description

This PR aims to get a few things in place for the dedicated setup flow. The flow isn't functional end to end yet, this just gets a few things in place, and the first view of it ready.

* Adds the new layout/getting started page
* Sets up logic to detect if we should be showing the flow. I anticipate this logic may need to change as we get further into the work and learn more, but for now this felt like a good place to start.
* Adds ability to force setup flow via query param `dedicated-setup=force`

For now, the new dedicated setup view will only show up under the following conditions:

* Query param set with `dedicated-setup=force`
* OR
* `enableDedicatedOnboardingFlow` flag set (can whitelist individual hosts) AND either no orgs or no auth providers are available.

![image](https://user-images.githubusercontent.com/367275/234999167-764d5c79-a2bc-441f-a0a2-e1ded39d6b1f.png)

## Related Issue(s)

Fixes WEB-243

## How to test

* One way to test this is to get an admin token for this preview environment and use that to log in.
  * w/ the preview env kube ctx loaded, run `previewctl admin credentials create`
  * take the token string output, and load the url w/ it, `/api/login/ots/admin/<token>`
* Another way to test is via starting the dashboard build w/ `yarn start` in a workspace against this branch. Set the `x-frontend-dev-url` header w/ your workspace url, then load up another env, like staging, and add `?dedicated-setup=force` to the url. This should also load the new view.
* If the view loads, then things are working as intended. There's no step beyond the getting started view yet in this PR.

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - bmh-dedica4d01d505b4</li>
	<li><b>🔗 URL</b> - <a href="https://bmh-dedica4d01d505b4.preview.gitpod-dev.com/workspaces" target="_blank">bmh-dedica4d01d505b4.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

</details>

<details>
<summary>Installer Options</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold
